packages/compiler/src/experimental/realm.ts

```diff
@@ -106,17 +127,51 @@ export class Realm {
    * Stores types that are deleted in this realm. When a realm is active and doing a traversal, you will
    * not find this type in e.g. collections. Deleted types are mapped to `null` if you ask for it.
    */
-  #deletedTypes = new Set<Type>();
+  #deletedTypes = new WeakSet<Type>();
 
   #stateMaps = new Map<symbol, Map<Type, any>>();
   public key!: symbol;
 
+  /**
+   * Create a new realm in the given program.
+   *
+   * @param program - The program to create the realm in.
+   * @param description - A short description of the realm's purpose.
+   */
   constructor(program: Program, description: string) {
     this.key = Symbol(description);
     this.#program = program;
+
     Realm.#knownRealms.set(this.key, this);
   }
 
+  #_typekit: Typekit | undefined;
```

Comment:
# and _ seems a bit extra?

Reply:
My thinking here was "# because I want it to be runtime-private" and "_ because I dont want the realm implementation to actually use this". `[REALM_TYPEKIT]` is the infallible symbol prop that should actually be used to access the cached typekit on the realm, and it's internal to the compiler.

There is at least some method to this #_ madness.

Reply:
I had to change away from using the REALM_TYPEKIT symbol anyway because TypeScript does not generate `declare const REALM_TYPEKIT: unique symbol;` and therefore the .d.ts is broken when you use an internal symbol as an accessor.